lib/auth/init.go

```diff
@@ -484,6 +483,49 @@ func Init(cfg InitConfig, opts ...ServerOption) (*Server, error) {
 	return asrv, nil
 }
 
+func initSetAuthPreference(asrv *Server, newAuthPref services.AuthPreference) error {
+	storedAuthPref, err := asrv.GetAuthPreference()
+	if err != nil {
+		if !trace.IsNotFound(err) {
+			return trace.Wrap(err)
+		}
+		storedAuthPref = nil
+	}
+	shouldReplace, err := shouldInitReplaceResource(storedAuthPref, newAuthPref)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if shouldReplace {
+		err = asrv.SetAuthPreference(newAuthPref)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		log.Infof("Updating auth preference: %v.", newAuthPref)
+	}
+	return nil
+}
+
+// shouldInitReplaceResource determines whether the candidate resource should
+// be used to replace the stored resource during auth server initialization.
```

Comment:
I would explain when this is the case and why in this godoc. We can keep the link to RFD but having to read the design document to understand why the function does what it does isn't very convenient IMO.